pkg/scm/scm.go

```diff
@@ -23,9 +23,13 @@ func DownloaderForSource(s string, forceCopy bool) (build.Downloader, string, er
 		return &empty.Noop{}, s, nil
 	}
 
-	details, mods := git.ParseFile(s)
+	details, mods, err := git.ParseFile(s)
 	glog.V(4).Infof("return from ParseFile file exists %v proto specified %v use copy %v", details.FileExists, details.ProtoSpecified, details.UseCopy)
+	if err != nil {
+		return nil, s, err
+	}
 
+	// NOTE: This could be handled by an error
```

Comment:
i think it's being handled correctly, it's up to the caller whether it wants to try to press on w/ the other git information returned and ignore the bad ref, or bail out.